tests/v2/validation/encryption/rke2_secret_encryption_test.go

```diff
@@ -0,0 +1,168 @@
+	require.NoError(s.T(), err)
+	logrus.Infof("secret created successfully!")
+
+	envFrom := []v1.EnvFromSource{
+		{
+			SecretRef: &v1.SecretEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{Name: secretObj.Name},
+			},
+		},
+	}
+	wload := namegen.AppendRandomString("test-workload")
+	containerTemplate := workloads.NewContainer("ranchertest", "ranchertest/mytestcontainer", v1.PullAlways, []v1.VolumeMount{}, envFrom)
+	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, nil)
+	deploymentTemplate := workloads.NewDeploymentTemplate(wload, s.ns, podTemplate, true, nil)
+
+	_, err = steveClient.SteveType(workloads.DeploymentSteveType).Create(deploymentTemplate)
+	require.NoError(s.T(), err)
+
+	podResults, podErrors := pods.StatusPods(client, clusterID)
+	assert.NotEmpty(s.T(), podResults)
+	assert.Empty(s.T(), podErrors)
+
+	kubeConfig, err := kubeconfig.GetKubeconfig(client, clusterID)
+	require.NoError(s.T(), err)
+
+	restConfig, err := (*kubeConfig).ClientConfig()
+	require.NoError(s.T(), err)
+
+	pods, err := steveClient.SteveType(pods.PodResourceSteveType).List(nil)
+	require.NoError(s.T(), err)
+
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		"echo $testKey",
+	}
+
+	for _, pod := range pods.Data {
+		if strings.HasPrefix(pod.ObjectMeta.Name, wload) {
+			err := kwait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+				output, err := kubeconfig.KubectlExec(restConfig, pod.ObjectMeta.Name, s.ns, cmd)
+				if err != nil {
+					return false, nil
+				}
+				require.Equal(s.T(), output.String(), secretValue)
+				logrus.Info("environment variable "+secretKey+" value : ", output.String())
+				return true, nil
+			})
+			require.NoError(s.T(), err)
+			break
+		}
+	}
+
+	nodesList, err := steveClient.SteveType("node").List(nil)
+	require.NoError(s.T(), err)
+	assert.NotEmpty(s.T(), nodesList)
+
+	_, err = client.ReLogin()
+	require.NoError(s.T(), err)
+
+	for _, node := range nodesList.Data {
+		result, err := strconv.ParseBool(node.Labels[nodes1.EtcdNodeLabel])
+		if result && node.Annotations[nodes1.ClusterAnnotation] == clusterName && err == nil {
+			machineName := node.Annotations[nodes1.MachineAnnotation]
+			sshKey, err := sshkeys.DownloadSSHKeys(client, machineName)
+			require.NoError(s.T(), err)
+			assert.NotEmpty(s.T(), sshKey)
+
+			sshUser := s.sshUser
+			assert.NotEmpty(s.T(), sshUser)
+
+			clusterNode := &nodes.Node{
+				NodeID:          node.ID,
+				PublicIPAddress: node.Annotations[nodes1.RKE2ExternalIP],
+				SSHUser:         sshUser,
+				SSHKey:          []byte(sshKey),
+			}
+			_, err = clusterNode.ExecuteCommand(EtcdCtlInstallCmd)
+			require.NoError(s.T(), err)
+
+			output, err := clusterNode.ExecuteCommand(fmt.Sprintf(rke2EtcdEncryptionCheckCmd, secretObj.Name))
+			require.NoError(s.T(), err)
+			require.NotEmpty(s.T(), output)
+			require.Contains(s.T(), output, "k8s:enc:aescbc")
```

Comment:
I don't think this check should be hard coded string. I had to look up what this meant; would it be appropriate to link to k8s documentation as to why it is done this way in our code? https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/#verifying-that-data-is-encrypted

Reply:
Yes, I've added this documentation as a comment in the code now.